lib/realtime/tenants/repo/migrations/20231204144024_create_rls_helper_functions.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Realtime.Tenants.Migrations.CreateRlsHelperFunctions do
+  @moduledoc false
+
+  use Ecto.Migration
+
+  def change do
+    execute("""
+    create or replace function realtime.channel_name() returns text as $$
+    select nullif(current_setting('realtime.channel_name', true), '')::text;
+    $$ language sql stable;
+    """)
+
+    execute("create schema if not exists auth")
+
+    execute("""
+    do $func_exists$
+    begin if not exists(select * from pg_proc where proname = 'role')
+    then
+      create or replace function auth.uid() returns uuid as $function_def$
```

Comment:
Why are we managing auth functions exactly again? These are not our fns.

Reply:
self hosting made easier basically. if we want to keep realtime as self deployable as possible we need to create them on our own if they are not present